tests/sentry/snuba/test_query_subscription_consumer.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import absolute_import
+
+import json
+import unittest
+from copy import deepcopy
+
+from exam import fixture, patcher
+from mock import Mock
+
+from sentry.snuba.models import QuerySubscription
+from sentry.snuba.query_subscription_consumer import (
+    InvalidMessageError,
+    InvalidSchemaError,
+    QuerySubscriptionConsumer,
+    register_subscriber,
+    subscriber_registry,
+)
+from sentry.testutils.cases import TestCase
+
+
+class BaseQuerySubscriptionTest(object):
+    @fixture
+    def consumer(self):
+        return QuerySubscriptionConsumer("hello")
+
+    @fixture
+    def valid_wrapper(self):
+        return {"version": 1, "payload": self.valid_payload}
+
+    @fixture
+    def valid_payload(self):
+        return {
+            "subscription_id": "1234",
+            "values": {"hello": 50},
+            "timestamp": 1235,
+            "interval": 5,
+            "partition": 50,
+            "offset": 10,
+        }
+
+    def build_mock_message(self, data):
+        message = Mock()
+        message.value.return_value = json.dumps(data)
+        return message
+
+
+class HandleMessageTest(BaseQuerySubscriptionTest, TestCase):
+    metrics = patcher("sentry.snuba.query_subscription_consumer.metrics")
+
+    def test_no_subscription(self):
+        self.consumer.handle_message(self.build_mock_message(self.valid_wrapper))
+        self.metrics.incr.assert_called_once_with(
+            "snuba_query_subscriber.subscription_doesnt_exist"
+        )
+
+    def test_subscription_not_registered(self):
+        sub = QuerySubscription.objects.create(
+            project=self.project,
+            type="unregistered",
+            subscription_id="an_id",
+            dataset="something",
+            query="hello",
+            aggregations=[],
+            time_window=1,
+            resolution=1,
+        )
+        data = self.valid_wrapper
+        data["payload"]["subscription_id"] = sub.subscription_id
+        self.consumer.handle_message(self.build_mock_message(data))
```

Comment:
Do you want to also assert that the callback was not called?

Reply:
That's tricky, because have no callback related to `unregistered` in `subscriber_registry` and so there's nothing that would be called. If we got to that part of the code it would just error.